core/matrix/ell.cpp

```diff
@@ -195,6 +177,20 @@ void Ell<ValueType, IndexType>::move_to(Csr<ValueType, IndexType>* result)
 }
 
 
+template <typename ValueType, typename IndexType>
+void Ell<ValueType, IndexType>::resize(dim<2> new_size, size_type max_row_nnz)
+{
+    if (this->get_size() != new_size ||
+        this->get_num_stored_elements_per_row() != max_row_nnz) {
```

Comment:
In case you need to fill a matrix data by yourself, or allocate the storage yourself, you also need to know how many nonzeros per row you have (this may also be obvious from context, e.g. stencil matrices) to do the correctly sized allocation.
What use case do you imagine where you don't know the matrix nonzero count, but `resize` is an issue?